common/djangoapps/student/views.py

```diff
@@ -1564,6 +1564,32 @@ def login_user(request, error=""):  # pylint: disable=too-many-statements,unused
         if user_found_by_email_lookup and LoginFailures.is_feature_enabled():
             LoginFailures.increment_lockout_counter(user_found_by_email_lookup)
 
+        # TODO: Remove Django 1.11 upgrade shim
+        # SHIM: In Django 1.10+ the user will not authenticate if it is not active (after the initial account creation,
+        # see NEW_USER_AUTH_BACKEND for how we handle that). This code is duplicated below for the < 1.10 case where
+        # the user is successfully authenticated, but not active. To remove this shim, remove the version check here
+        # and the duplicated code at the end of this function.
+        #
+        # Note that this will be triggered for ALL inactive users, regardless of whether the password is correct. In
+        # either case they need to activate before continuing so password status is moot.
+        if django.VERSION >= (1, 10) and user_found_by_email_lookup and not user_found_by_email_lookup.is_active:
+            if settings.FEATURES['SQUELCH_PII_IN_LOGS']:
+                AUDIT_LOG.warning(
+                    u"Login failed - Account not active for user.id: {0}, resending activation".format(
+                        user_found_by_email_lookup.id)
+                )
+            else:
+                AUDIT_LOG.warning(u"Login failed - Account not active for user {0}, resending activation".format(
+                    user_found_by_email_lookup.username)
+                )
+
+            reactivation_email_for_user(user_found_by_email_lookup)
+
+            return JsonResponse({
+                "success": False,
+                "value": _generate_not_activated_message(user_found_by_email_lookup),
+            })  # TODO: this should be status code 400  # pylint: disable=fixme
```

Comment:
Dang it. I just spent some time on a solution for this issue as well:
https://github.com/edx/edx-platform/pull/17130/commits/47c48c6b261f2d443cd2485673b4fd62a17fa7c2
But this solution seems more targeted and therefore better.

Reply:
I will now check this PR before attempting any further LMS shard 4 fixes - lesson learned!